technology-considerations/TDR Database Design.md

```diff
@@ -0,0 +1,320 @@
+# TDR Database Design
+
+
+## Summary
+
+
+This document aims to recommend a suitable database schema, for the TDR
+application. Consideration is given to a design which can evolve to
+support enhanced metadata capture without introducing an over-prescribed
+or rigid nature that may limit the flexibility for use as TDR evolves.
+
+## Scope
+
+
+The Transfer of Digital Records (TDR) application allows a user of an
+organisation to transfer digital records directly to TNA. A database
+schema is required to store information to support the TDR application
+process and the creation of an export package. An additional requirement
+is that the database itself may not to be used for long term storage of
+the information and is likely to be purged at intervals of time or after
+a transfer has occurred.
+
+The TDR application will have enhanced functionality delivered across
+multiple phases releases, consideration for which should be included in
+the design of the database schema. Consideration of this at an early
+stage will help minimise database maintenance and code changes required
+for each of the releases whilst still ensuring data capability.
+
+## Database Design
+
+
+The following sections propose an initial design for the TDR database
+schema. The database design may change in the future to reflect
+requirement changes within the UI and / or export package needs.
+
+## Overall Schema
+
+
+The overall schema proposed is shown below
+
+![](./images/databaseschema.jpeg)
+
+The following sections detail the tables proposed.
+
+### User
+
+The suggested User table is shown below.
+
+In this table the user is associated with some basic information and a
+body. The body is their working organisation and is known as the
+"Transferring Body".
+
+Security and user permissions are believed to be enforced in the
+application through AWS IAM policy implementations. If this should
+change the User table may be modified to support the features required.
+
+At this stage it is not suggested to add audit information such as
+lastLogon date to this table but can be extended if required.
+
+| Field | Description | Data Value type | Key Type | Comment |
+| ----- | ----- | ----- | ----- | ----- |
+| UserId | The internal ID of the User  | | PK             
+| BodyId | The ID of the Body the user is related to | | FK | Transferring Body. Assuming a user only has one organisational Body they work for.
+| Name | The Name of the user | String | | We could separate first and lastname
+| Email | The email address of the user | String                               
+| PhoneNo | The phone number of the user | String                               
```

Comment:
Out of curiosity, why do we need a user's phone number?

Reply:
It's a good question. It's more for later come back as the archivists may need to contact them and I know they do sometimes call them to discuss things through. It was me who added it rather than a business discussion. I didn't see any harm in it. but it would need to be added to the user creation page.

Reply:
oh and it is still a proposal subject to change so it could easily be dropped in the app.

Reply:
David's been looking into data protection for TDR, so let's leave this in here and we'll change it if necessary. :+1: